### PR TITLE
feat: Added API for write completion response to TSDB.

### PR DESCRIPTION
**Pull request by takeshinoda**

### CODE REVIEW

File: mqtt.go [hunk 1]
```diff
@@ -28,24 +28,25 @@ type (
...
-		wgRead        sync.WaitGroup
+		conn           net.Conn
+		stream         *packet.Stream
+		dc             *DeviceContext
+		packetID       uint16
+		subs           map[string]mqttSubscription
+		command        chan<- *DeviceCommand
+		event          <-chan *DeviceEvent
+		bulkData       <-chan *DeviceBulkData
+		syncedBulkData <-chan *DeviceSyncedBulkData
```
Comment (takeshinoda) [inline mqtt.go]: Added line

File: session.go [hunk 2]
```diff
@@ -14,12 +14,13 @@ type (
...
-		cmdQueue      chan *DeviceCommand
-		evtQueue      chan *DeviceEvent
-		bulkDataQueue chan *DeviceBulkData
-		kvCache       *keyValueCache
+		dc                  *DeviceContext
+		conn                connection
+		cmdQueue            chan *DeviceCommand
+		evtQueue            chan *DeviceEvent
+		bulkDataQueue       chan *DeviceBulkData
+		syncedBulkDataQueue chan *DeviceSyncedBulkData
```
Comment (takeshinoda) [inline session.go]: Added line

File: session.go [hunk 3]
```diff
@@ -92,13 +98,14 @@ var (
...
-		sendEvent    chan *sessCtrlSendEvent
-		sendBulkData chan *sessCtrlSendBulkData
-		accessKV     chan *sessCtrlAccessKV
-		ping         <-chan time.Time
+		start         chan *sessCtrlStart
+		stop          chan *sessCtrlStop
+		getState      chan *sessCtrlGetState
+		sendEvent     chan *sessCtrlSendEvent
+		sendBulkData  chan *sessCtrlSendBulkData
+		writeBulkData chan *sessCtrlWriteBulkData
```
Comment (takeshinoda) [inline session.go]: Added line

**Review by honteng — COMMENTED** [review]
It looks good to me. Ask @ethankan for the final approval.

**Review by ethankan — COMMENTED** [review]
@takeshinoda Since there is already a "BulkData" API, can you explain what is different in the "SyncedBulkData" API?

**Comment by takeshinoda** [comment]
@ethankan 
Since there is already API asynchronous (the following part), so added a synchronous API.

https://github.com/moderepo/device-sdk-go/pull/17/files#diff-f2fccfb96064097b73fbe8eeae3703eeR412

**Review by ethankan — COMMENTED** [review]
The "syncedBulkDataQueue" has a length of 128. But since it's the call is supposed to be synchronous there is no need for a "queue".

Anyway, I'm still not 100% sure the SDK should have a mixture of sync and async publishing. I'm also not fully satisfied with the existing queue implementation (back pressure is not handled well). I wonder if we should create a SDK 2.0 that is simpler and more robust. Let me discuss with Gaku and Naoki a little bit. Thanks.

**Comment by takeshinoda** [comment]
@ethankan 
Thanks.

I think so too, about no need queue. So, I will delete about queue implementation.

>  I'm still not 100% sure the SDK should have a mixture of sync and async publishing. 

I think SDK should receive a mixed AckPacket of Sync and Asyc, or separate connections. We need discuss about this issue. but, May I proceed with the current major version implementation here?

**Review by ethankan — APPROVED** [review]
Actually GO SDK is currently on v2.x. Let's add this feature to it.

We shall discuss what we should do in "v3" later.

**Comment by takeshinoda** [comment]
OK, thanks!

**Comment by takeshinoda** [comment]
I will bump up version after long run test.

**Pull request merged**